nats/aio/msg.py

```diff
@@ -14,17 +14,27 @@
 
 import datetime
 from dataclasses import dataclass
+from typing import TYPE_CHECKING, Dict, List, Optional
 from nats.errors import Error, NotJSMessageError, MsgAlreadyAckdError
 
+if TYPE_CHECKING:
+    from nats import NATS
 
+
+@dataclass
 class Msg:
     """
     Msg represents a message delivered by NATS.
     """
-    __slots__ = (
-        'subject', 'reply', 'data', 'sid', '_client', 'headers', '_metadata',
-        '_ackd'
-    )
+    client: "NATS"
```

Comment:
I think fine either way, but why the rename of `_client` into `client`?

Reply:
Dataclasses don't allow you to underscore whatever fields you want to pass in `__init__`. You either keep it internal-only ("private", in some sense) or make it available for bot outside initialization and outside access. If we decide to go with the old approach and drop dataclasses, I'll revert it back as well.

Reply:
I'd rather keep it 'private' as `_client` in this case, maybe just set it outside of `build_message` and remove it as a parameter for `__init__`?
```
diff --git a/nats/aio/client.py b/nats/aio/client.py
index a41bfb9..e42f348 100644
--- a/nats/aio/client.py
+++ b/nats/aio/client.py
@@ -1413,6 +1413,7 @@ class Client:
                 return
 
         msg = self._build_message(subject, reply, data, hdr)
+        msg._client = self
         if not msg:
             return
 
@@ -1513,7 +1514,6 @@ class Client:
             reply=reply.decode(),
             data=data,
             headers=headers,
-            client=self
         )
```

Reply:
Great news! I was wrong. Turned out, you can pass underscored attributes into a dataclass. Did I confuse it with [attrs](https://attrs.org/)? 🤔 Anyway, I've fixed it now.